src/main/java/org/assertj/core/internal/Classes.java

```diff
@@ -250,7 +250,7 @@ public void assertIsNotFinal(AssertionInfo info, Class<?> actual) {
    * @throws AssertionError if the actual {@code Class} doesn't contains all of these annotations.
    */
   public void assertContainsAnnotations(AssertionInfo info, Class<?> actual,
-                                        @SuppressWarnings("unchecked") Class<? extends Annotation>... annotations) {
+                                        Class<? extends Annotation>... annotations) {
```

Comment:
removing this leads to a "Type safety: Potential heap pollution via varargs parameter annotations" warning, adding `@SafeVarargs` is possible here since this is not proxyed or subclassed.

Reply:
For me the `SuppressWarnings` have no influence on the possible heap pollution (when compiling with `-Xlint:unchecked` in IntelliJ). 

We can't make it `final` because `Classes` is mocked in the tests. If we make it final tests start to fail (already tried that 😄 ). However, what we can do is to use an array instead of varargs. That's how I got rid of those errors in the `ForProxy` methods.

Reply:
Another solution would be to annotated the method with `@SuppressWarnings("unchecked")`. The heap pollution is gone for me with that.

Reply:
Using an array is the best solution IMO, varargs are great for user facing APIs but for internal code arrays are fine.